test/libsolidity/SolidityEndToEndTest.cpp

```diff
@@ -10589,6 +10589,535 @@ BOOST_AUTO_TEST_CASE(snark)
+	contract test {
+		ufixed a = 3.5;
+		ufixed b = 1.25;
+
+		function f() public returns (ufixed) {
+			return a - b;
+		}
+	}
+	)";
+	compileAndRun(sourceCode, 0, "test");
+	ABI_CHECK(callContractFunction("f()"), encodeArgs(createFixed(225, 100, 19)));
+}
+
+BOOST_AUTO_TEST_CASE(fixed_variable_subtraction)
+{
+	char const* sourceCode = R"(
+	contract test {
+		fixed a = 3.5;
+		fixed b = 1.25;
+
+		function f() public returns (fixed) {
+			return b - a;
+		}
+	}
+	)";
+	compileAndRun(sourceCode, 0, "test");
+	ABI_CHECK(callContractFunction("f()"), encodeArgs(createFixed(-225, 100, 19)));
+}
+
+BOOST_AUTO_TEST_CASE(ufixed_constant_multiplication)
+{
+	char const* sourceCode = R"(
+	contract test {
+		function f() public returns (ufixed) {
+			return 3.5 * 1.25;
+		}
+	}
+	)";
+	compileAndRun(sourceCode, 0, "test");
+	ABI_CHECK(callContractFunction("f()"), encodeArgs(createFixed(4375, 1000, 19)));
+}
+
+BOOST_AUTO_TEST_CASE(fixed_constant_multiplication)
+{
+	char const* sourceCode = R"(
+	contract test {
+		function f() public returns (fixed) {
+			return 3.5 * -1.25;
+		}
+	}
+	)";
+	compileAndRun(sourceCode, 0, "test");
+	ABI_CHECK(callContractFunction("f()"), encodeArgs(createFixed(-4375, 1000, 19)));
+}
+
+BOOST_AUTO_TEST_CASE(ufixed_variable_multiplication)
+{
+	char const* sourceCode = R"(
+	contract test {
+		ufixed a = 3.5;
+		ufixed b = 1.25;
+
+		function f() public returns (ufixed) {
+			return a * b;
+		}
+	}
+	)";
+	compileAndRun(sourceCode, 0, "test");
+	ABI_CHECK(callContractFunction("f()"), encodeArgs(createFixed(4375, 1000, 19)));
+}
+
+BOOST_AUTO_TEST_CASE(fixed_variable_multiplication)
+{
+	char const* sourceCode = R"(
+	contract test {
+		fixed a = 3.5;
+		fixed b = -1.25;
+
+		function f() public returns (fixed) {
+    		return a * b;
+		}
+	}
+	)";
+	compileAndRun(sourceCode, 0, "test");
+	ABI_CHECK(callContractFunction("f()"), encodeArgs(createFixed(-4375, 1000, 19)));
+}
+
+BOOST_AUTO_TEST_CASE(ufixed_constant_division)
+{
+	char const* sourceCode = R"(
+	contract test {
+		function f() public returns (ufixed) {
+			return 3.5 / 1.25;
+		}
+	}
+	)";
+	compileAndRun(sourceCode, 0, "test");
+	ABI_CHECK(callContractFunction("f()"), encodeArgs(createFixed(28, 10, 19)));
+}
+
+BOOST_AUTO_TEST_CASE(fixed_constant_division)
+{
+	char const* sourceCode = R"(
+	contract test {
+		function f() public returns (fixed) {
+			return 3.5 / -1.25;
+		}
+	}
+	)";
+	compileAndRun(sourceCode, 0, "test");
+	ABI_CHECK(callContractFunction("f()"), encodeArgs(createFixed(-28, 10, 19)));
+}
+
+
+BOOST_AUTO_TEST_CASE(ufixed_variable_division)
+{
+	char const* sourceCode = R"(
+	contract test {
+		ufixed a = 3.5;
+		ufixed b = 1.25;
+
+		function f() public returns (ufixed) {
+			return a / b;
+		}
+	}
+	)";
+	compileAndRun(sourceCode, 0, "test");
+	ABI_CHECK(callContractFunction("f()"), encodeArgs(createFixed(28, 10, 19)));
+}
+
+BOOST_AUTO_TEST_CASE(fixed_variable_division)
+{
+	char const* sourceCode = R"(
+	contract test {
+		fixed a = 3.5;
```

Comment:
Before we merge this, I think we should make `ufixed` be a `ufixed128x18`, so that `1 ether` can be stored as `1` in a `ufixed`

Reply:
I can make that change, I just couldn't find where the default is in the code.

Reply:
Types.cpp:200

Reply:
Nevermind, found where the default is.